### PR TITLE
feat: Dark mode implemented

### PR DESCRIPTION
**Pull request by psankhe28**
### What

Implemented a dark mode toggle.

### Screenshot

By default: Adapts to the system theme (dark mode for me).

image

Light mode:

image

### Part of 

[#274](https://github.com/openfoodfacts/open-prices-frontend/issues/274)

### CODE REVIEW

File: src/components/Header.vue [hunk 1]
```diff
@@ -1,6 +1,12 @@
 <template>
   <v-app-bar :elevation="1">
     <v-app-bar-nav-icon @click.stop="showDrawerMenu = !showDrawerMenu"></v-app-bar-nav-icon>
+    <div style="margin-top: 8px; margin-right: 16px;">
```
Comment (raphodn) [inline src/components/Header.vue:4]: why a custom component ? we use vuetify everywhere, they have a [switch component](https://vuetifyjs.com/en/components/switches/#usage)

**Review by raphodn — COMMENTED** [review]
hi psankhe28, thanks for your first PR ! a few comments to try and simplify :)

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
   "dependencies": {
     "@intlify/unplugin-vue-i18n": "^2.0.0",
     "@vue-leaflet/vue-leaflet": "^0.10.1",
-    "@vueuse/core": "^10.6.1",
+    "@vueuse/core": "^10.7.2",
```
[the remark this answers is not in the document]
Comment (psankhe28) [reply package.json:15]: Okay, I will change it back to as it was.

File: src/components/Footer.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-footer class="bg-grey-lighten-1" style="max-height: 20%">
+  <v-footer class="bg-grey-lighten-1" style="position: absolute; bottom: -0; width: 100%;overflow: hidden;">
```
[the remark this answers is not in the document]
Comment (psankhe28) [reply src/components/Footer.vue:2]: After incorporating the toggle, the footer experienced overflow issues, so I have resolved the issue using this styles.

File: src/components/Header.vue (hunk 1 above)
Comment (psankhe28) [reply src/components/Header.vue:4]: Alright, I'll adjust the position by placing it on the far right of the navigation bar. Additionally, I wasn't aware of the custom component, but I'll make the required modifications.

Comment (teolemon) [reply src/components/Header.vue:4]: In Settings, honestly. Honor the browser default, and memorized in settings

Comment (psankhe28) [reply src/components/Header.vue:4]: Sorry @teolemon, but I didn't understand what does that mean? Can you please elaborate?

Comment (teolemon) [reply src/components/Header.vue:4]: 1. show the Dark mode settings in the settings screen
2. 3 Options: Dark mode, Light Mode, Follow system settings
3. By default, it follows system settings, as exposed by the browser https://css-tricks.com/a-complete-guide-to-dark-mode-on-the-web/

Comment (psankhe28) [reply src/components/Header.vue:4]: Okay, I will make the necessary changes. Thanks for elaborating!

**Comment by psankhe28** [comment]
"Apologies, @raphodn, I accidentally force-pushed. Additionally, I'm sorry for the delayed contribution to this pull request due to academic commitments. I have a question regarding the toggle button, as pointed out by @teolemon. It's supposed to be in the settings, but I'm having trouble locating it. Can you provide some guidance on this matter?"

**Comment by dq18** [comment]
Hi @psankhe28 the user settings vue page is here:
https://github.com/openfoodfacts/open-prices-frontend/blob/master/src/views/UserSettings.vue

**Pull request closed without merging**